src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -44,6 +45,15 @@ public function process(ContainerBuilder $container)
                     $this->completeDefinition($id, $definition);
                 }
             }
+
+            foreach ($this->usedTypes as $type => $id) {
+                if (isset($this->usedTypes[$type]) && isset($this->notGuessableTypes[$type])) {
+                    $classOrInterface = class_exists($type) ? 'class' : 'interface';
+                    $matchingServices = implode(', ', $this->types[$type]);
+
+                    throw new RuntimeException(sprintf('Unable to autowire argument of type "%s" for the service "%s". Multiple services exist for this %s (%s).', $type, $id, $classOrInterface, $matchingServices));
```

Comment:
Only the last service in which the type is used will be shown, would you prefer a list of all of them?